src/components/calcite-input/calcite-input.tsx

```diff
@@ -425,14 +442,14 @@ export class CalciteInput {
       }
       return;
     }
-    if (event.key == getGroupSeparator(this.locale)) {
-      return;
-    }
-    if (
-      event.key == getDecimalSeparator(this.locale) &&
-      this.localizedValue.indexOf(event.key) === -1
-    ) {
-      return;
+    const decimalSeparator = getDecimalSeparator(this.locale);
+    if (event.key === decimalSeparator) {
+      if (!this.value && !this.childNumberEl.value) {
+        return;
+      }
+      if (this.value && this.childNumberEl.value?.indexOf(decimalSeparator) === -1) {
```

Comment:
Should this be checking `localizedValue` instead of going through `childNumberEl.value`?

Reply:
Nope.  It needs to check `childNumberEl.value` because of the case when a user types in `1234.` which will not update `this.value` to `1234.` because that is an invalid number, so in order to allow typing decimal characters we have to check the actual value in the input instead of `localizedValue` or `value`.  This logic also prevents the user from typing more than 1 decimal character in the input.

Reply:
Gotcha.

One more thing I spotted here: `value` is guaranteed here: `this.childNumberEl.value`